crates/red_knot_python_semantic/resources/mdtest/intersection_types.md

```diff
@@ -635,6 +635,68 @@ def _(
     reveal_type(i8)  # revealed: Never
 ```
 
+### Simplifications of `bool`, `AlwaysTruthy` and `AlwaysFalsy`
+
+In general, intersections with `AlwaysTruthy` and `AlwaysFalsy` cannot be simplified. Naively, you
+might think that `int & AlwaysFalsy` could simplify to `Literal[0]`, but this is not the case: for
+example, the `False` constant inhabits the type `int & AlwaysFalsy` (due to the fact that
+`False.__class__` is `bool` at runtime, and `bool` subclasses `int`), but `False` does not inhabit
+the type `Literal[0]`.
+
+Nonetheless, intersections of `AlwaysFalsy` or `AlwaysTruthy` with `bool` _can_ be simplified, due
+to the fact that `bool` is a `@final` class at runtime that cannot be subclassed.
+
+```py
+from knot_extensions import Intersection, Not, AlwaysTruthy, AlwaysFalsy
+
+def f(
```

Comment:
You *could* write all of the tests in this test suite using
```py
static_assert(is_equivalent_to(Intersection[bool, AlwaysTruthy], Literal[True]))
```
because they're all fully-static types. I think it makes them easier to read. On the other hand, it would make them dependent on a correct implementation of `is_equivalent_to`, so feel free to leave them in the current form if you think it makes them more independent.

Reply:
I think I weakly prefer the tests the way they are, because I don't just want to test that `bool & AlwaysTruthy` is _equivalent_ to `Literal[True]` (that could be done with a special case in `Type::is_equivalent_to`); I want to test that `bool & AlwaysTruthy` is _eagerly simplified_ to `Literal[True]`.

Do my tests actually test that? Well... no, they actually test that the intersection is simplified to a type that `Type::display` displays as `"Literal[True]"`. The type could always be special-cased in `Type::display`; all mdtests are to some extent integration tests rather than unit tests if we're being pedantic about it. But it still feels slightly _closer_ to what I'd _like_ to test here to keep them as they are.